### DISCUSSION TITLE
How to format DT_DURATION_xxx ?

### DISCUSSION THREAD

**Discussion by Schnippsche** [ezylang/EvalEx#399] [Q&A]
Hi,
at first thanks for the wonderful library...

I tested the 3.1.0 Snapshot because i need the date time features. It works very well but now i didn't know how i can format a Duration.

This works:
```
Expression expression1 = new Expression("DT_DURATION_MILLIS(ts)").with("ts", 50400000);
System.out.println(expression1.evaluate().getStringValue());
```
This gives me "PT14H" which is correct. 

Now i try to format the expression with DT_FORMAT() 
```
Expression expression2 = new Expression("DT_FORMAT(DT_DURATION_MILLIS(ts), \"HH:mm:ss\")").with("ts", 50400000);
System.out.println(expression2.evaluate().getStringValue());
```
This gives me "01:00:00"
are there plans to support formatting of Durations ?

Thanks
Stefan

**Comment by uklimaschewski** [comment]
First of all, be careful when using the snapshot. The date/time function will change a bit and the current functions are not compatible with what will be in the final 3.1.0 release. I will also rename some of them.

Java has no built-in function to format a duration, as far as I know.
The formatter in Java we use needs some kind of temporal object, e.g. a _Date_ instance.
A simple solution would be to use a date of of 0/0/0 00:00 and add the duration, the resulting date could be formatted.
But I think this is reliable only for short durations less than a day or so. Tricky part would be daylight savings and leap years, I think.

**Comment by Schnippsche** [comment]
Thank you for your answer; my project is still under development and i can wait for the official 3.1. release. 
Is there already a schedule for the release?
Stefan

**Reply by uklimaschewski** [reply]
This is a private project, there is no schedule. It highly depends on the time I have beside my work and other activities.
I would estimate a few more days of work to finish 3.2, hopefully I will have some time in the next few weeks.
